src/App.js

```diff
@@ -0,0 +1,108 @@
+import './App.css';
+import {Checklist} from './Checklist.js'
+import {AddTaskPage} from "./AddTaskPage";
+import {useEffect, useState} from "react";
+import {IncompleteTasksOnly} from "./IncompleteTasksOnly";
+import {generateUniqueID} from "web-vitals/dist/modules/lib/generateUniqueID";
+import {ModifyTaskPage} from "./ModifyTaskPage";
+
+export function App(props) {
+
+    let [data, setData] = useState(props.initialData);
+    let [itemsToShow, setItemsToShow] = useState("both");
+    const initialCompletedItems = data.map((item) => item.completed ? item.id : undefined);
+    let [completedItems, setCompletedItems] = useState(initialCompletedItems);
+    let [componentsToRender, setComponentsToRender] = useState(<Checklist items={data} handleChangeField={handleChangeField} modifyTask={modifyTask} completedItems={completedItems} changeCompletedItems={setCompletedItems}/>);
+
+    let taskButtonValue = (itemsToShow === "both" ? "Hide Completed Tasks" : "Show Completed Tasks");
+    let [showButtons, setShowButtons] = useState(true);
+
+    useEffect(() => {
+        setComponentsToRender(<Checklist items={data} handleChangeField={handleChangeField} modifyTask={modifyTask} completedItems={completedItems} changeCompletedItems={setCompletedItems}/>);
+        setShowButtons(true);
+    }, [data]);
+
+    useEffect(() => {
+        setComponentsToRender(<Checklist items={data} handleChangeField={handleChangeField} modifyTask={modifyTask} completedItems={completedItems} changeCompletedItems={setCompletedItems}/>);
+        setShowButtons(true);
+    }, [completedItems]);
+
+    function addNewDataPoint(e, value) {
+        e.preventDefault();
+        let newDataMember = {
+            title: value,
+            id: generateUniqueID(),
+            completed: false,
+        }
+        let newData = [...data, newDataMember];
+        setData(newData);
+    }
+
+    function handleChangeField(e, id, field, value) {
+        e.preventDefault();
```

Comment:
Same, no need for e. If you use a button input rather than a submit input, there's no need for the preventDefault.